lens-cube/src/main/java/org/apache/lens/cube/parse/CandidateUtil.java

```diff
@@ -205,4 +218,75 @@ public int compare(UnionCandidate o1, UnionCandidate o2) {
       return Integer.valueOf(o1.getChildren().size() - o2.getChildren().size());
     }
   }
+
+  private static final String baseQueryFormat = "SELECT %s FROM %s";
+
+  public static String createHQLQuery(String select, String from, String where, String groupby, String orderby, String having,
+                                      Integer limit) {
+
+    List<String> qstrs = new ArrayList<String>();
+    qstrs.add(select);
+    qstrs.add(from);
+    if (!StringUtils.isBlank(where)) {
+      qstrs.add(where);
+    }
+    if (!StringUtils.isBlank(groupby)) {
+      qstrs.add(groupby);
+    }
+    if (!StringUtils.isBlank(having)) {
+      qstrs.add(having);
+    }
+    if (!StringUtils.isBlank(orderby)) {
+      qstrs.add(orderby);
+    }
+    if (limit != null) {
+      qstrs.add(String.valueOf(limit));
+    }
+
+    StringBuilder queryFormat = new StringBuilder();
+    queryFormat.append(baseQueryFormat);
+    if (!StringUtils.isBlank(where)) {
+      queryFormat.append(" WHERE %s");
+    }
+    if (!StringUtils.isBlank(groupby)) {
+      queryFormat.append(" GROUP BY %s");
+    }
+    if (!StringUtils.isBlank(having)) {
+      queryFormat.append(" HAVING %s");
+    }
+    if (!StringUtils.isBlank(orderby)) {
+      queryFormat.append(" ORDER BY %s");
+    }
+    if (limit != null) {
+      queryFormat.append(" LIMIT %s");
+    }
+    return String.format(queryFormat.toString(), qstrs.toArray(new String[0]));
+  }
+
+  public static void updateFinalAlias(ASTNode selectAST, CubeQueryContext cubeql) {
+    for (int i = 0; i < selectAST.getChildCount(); i++) {
+      ASTNode selectExpr = (ASTNode) selectAST.getChild(i);
+      ASTNode aliasNode = HQLParser.findNodeByPath(selectExpr, Identifier);
+      String finalAlias = cubeql.getSelectPhrases().get(i).getFinalAlias().replaceAll("`", "");
+      String actualAlias = cubeql.getSelectPhrases().get(i).getActualAlias();
+      if (actualAlias == null && selectExpr.getChildCount() == 2 ) {
```

Comment:
Update logic 

     if (actualAlias == null ) {
        if (aliasNode != null){
          //Since actual alias supplied by user is null, we should delete this alias node
          selectExpr.deleteChild(1);
        }
        continue;
      }